go/runtime/client/watcher.go

```diff
@@ -166,8 +179,18 @@ func (w *blockWatcher) watch() {
 
 				// Tell every client to resubmit as nothing further can be finalized by this committee.
 				for key, watch := range w.watched {
+					var epochNumber api.EpochTime
+					epochNumber, err = w.common.consensus.EpochTime().GetEpoch(w.common.ctx, height)
```

Comment:
You should create a function-local variable called `currentEpoch` and update it once above (e.g., immediately after entering the epoch transition block `if`).